src/uct/base/uct_iface.c

```diff
@@ -1034,6 +1037,163 @@ int uct_iface_local_is_reachable(uct_iface_local_addr_ns_t *addr_ns,
     return 1;
 }
 
+struct route_info {
+    int if_index;
+    int family;
+
+    union {
+        struct in_addr  ipv4;
+        struct in6_addr ipv6;
+    } remote_addr;
+
+    int prefix_len;
+    int reachable;
+};
+
+static int netlink_get_route_info(int **if_idx, void **dst_in_addr,
+                                  struct rtattr *rta, int len)
+{
+    *if_idx = NULL;
+    *dst_in_addr = NULL;
+
+    for (; RTA_OK(rta, len); rta = RTA_NEXT(rta, len)) {
+        if (rta->rta_type == RTA_OIF) {
+            *if_idx = RTA_DATA(rta);
+        }
+        else if (rta->rta_type == RTA_DST) {
+            *dst_in_addr = RTA_DATA(rta);
+        }
+    }
+
+    return 0;
```

Comment:
Why not void?

Reply:
Maybe better make it return UCS status, then don't need to check for NULL outside.

Reply:
@gleon99 But then I need to check the status outside of the function anyway, and I am also not sure if it's an error more than just an entry with missing properties. Which error status would you return then?